src/Lecture3.hs

```diff
@@ -125,10 +143,10 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  a <> b = Reward (rewardGold a <> rewardGold b) (getAny (Any (rewardSpecial a && rewardSpecial b)))
```

Comment:
I think the part with `getAny` and `Any` is redundant because you put the boolean value inside the `Any` constructor and then immediately unwrap it. So this does nothing.

Also, the correct implementation should use `||` and not `&&` because according to the task text, the final reward should be special if **either** reward is special and not **both** 🙂 

Probably, a separate unit test can be added to the course test suite to check this implementation 🙂

Reply:
I'm glad we found a special case for next courses 😅
And will fix.